Xamarin.Forms.Core/TemplateUtilities.cs

```diff
@@ -58,9 +58,9 @@ public static void OnContentChanged(BindableObject bindable, object oldValue, ob
 			var newElement = (Element)newValue;
 			if (self.ControlTemplate == null)
 			{
-				for (var i = 0; i < self.InternalChildren.Count; i++)
+				while (self.InternalChildren.Count > 0)
 				{
-					self.InternalChildren.Remove(self.InternalChildren[i]);
+					self.InternalChildren.RemoveAt(0);
```

Comment:
This makes the clearing an `O(n^2)` operation. Changing t his to `self.InternalChildren.RemoveAt(self.InternalChildren.Count - 1)` would make it O(n).